public/app/features/alerting/unified/utils/misc.ts

```diff
@@ -260,7 +260,10 @@ export function isErrorLike(error: unknown): error is Error {
 }
 
 export function getErrorCode(error: Error): unknown {
-  return isApiMachineryError(error) ? error.data.details.uid : error.cause;
+  if (isApiMachineryError(error) && error.data.details) {
```

Comment:
Could `isApiMachineryError` also check for presence of `.data.details`, and then we don't have to check it in the two `if` conditionals? So TypeScript sorts it for us instead?

Reply:
That's also my understanding. It might still be `ApiMachineryError` even without `details` field